Marlin/src/MarlinCore.cpp

```diff
@@ -1131,6 +1131,10 @@ inline void tmc_standby_setup() {
  *  - Set Marlin to RUNNING State
  */
 void setup() {
+  #ifdef HAL_STM32
+    FastIO_init();
+  #endif
```

Comment:
we can have FASTIO_INIT macro pointing to `FastIO_init` function and use `#TERN_(FASTIO_INIT, FASTIO_INIT())`, so any hal can define it in the future

Reply:
i prefer to let @thinkyhead review that an maybe clean the STM32 HAL... cf Marlin# git grep -n ARDUINO_ARCH_STM32 | grep MAPLE
```
Marlin/src/HAL/STM32/HAL.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/HAL.cpp:168:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/HAL_MinSerial.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/HAL_SPI.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/HAL_SPI.cpp:227:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/MarlinSPI.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1) && !defined(STM32H7xx)
Marlin/src/HAL/STM32/MarlinSPI.cpp:168:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1 && !STM32H7xx
Marlin/src/HAL/STM32/MarlinSerial.cpp:19:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/MarlinSerial.cpp:104:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/Sd2Card_sdio_stm32duino.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/Sd2Card_sdio_stm32duino.cpp:323:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/Servo.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/Servo.cpp:110:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/eeprom_flash.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/eeprom_flash.cpp:273:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/eeprom_sdcard.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/eeprom_sdcard.cpp:91:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/eeprom_sram.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/eeprom_sram.cpp:68:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/eeprom_wired.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/eeprom_wired.cpp:78:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/fast_pwm.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/fast_pwm.cpp:59:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/fastio.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/fastio.cpp:34:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/msc_sd.cpp:16:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/msc_sd.cpp:128:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/tft/gt911.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/tft/gt911.cpp:202:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/tft/tft_fsmc.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/tft/tft_fsmc.cpp:181:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/tft/tft_ltdc.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/tft/tft_ltdc.cpp:387:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/tft/tft_spi.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/tft/tft_spi.cpp:243:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/tft/xpt2046.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/tft/xpt2046.cpp:170:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/timers.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/timers.cpp:322:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/usb_host.cpp:23:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/usb_host.cpp:117:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/usb_serial.cpp:19:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/usb_serial.cpp:54:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/HAL/STM32/watchdog.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/HAL/STM32/watchdog.cpp:49:#endif // ARDUINO_ARCH_STM32 && !STM32GENERIC && !MAPLE_STM32F1
Marlin/src/lcd/extui/ftdi_eve_touch_ui/ftdi_eve_lib/compat.h:291:    #if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
Marlin/src/lcd/extui/mks_ui/wifiSerial_STM32.cpp:22:#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
ini/stm32f1-maple.ini:29:  ${common.build_flags} -DARDUINO_ARCH_STM32 -DMAPLE_STM32F1

```

Reply:
no hurry, that can wait days or weeks ;)

Reply:
can be simplified to #ifdef HAL_STM32 now

Reply:
most are just for Arduino IDE builds...

Reply:
i meant, each HAL has an unique define... except STM32 one... to include HAL/ cpp files or not

Reply:
When Marlin is compiled in Arduino IDE we want to make sure only the needed files get compiled. This clause looks like it might be compatible with some STM32 framework that you can use with Arduino IDE. If that is not the case, and and `HAL/STM32` can only ever be compiled by PlatformIO IDE, then this clause can be replaced everywhere in Marlin with `#ifdef HAL_STM32`.

Reply:
hmm, STM32GENERIC seems a third framework https://github.com/danieleff/STM32GENERIC, but our current STM32 frameworks dont have this word/define. May be outdated now there is a working alternative solution, to confirm...

Reply:
@rhapsodyv I didnt ignore your idea to allow fastio on other archs, but its STM32 HAL specific for now, if another one do it.. could be changed easily and could even be optional... to compare perfs of the feature...